src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1302,15 +1302,23 @@ trait Contexts { self: Analyzer =>
     // have to pass in context because multiple contexts may share the same ReportBuffer
     def reportFirstDivergentError(fun: Tree, param: Symbol, paramTp: Type)(implicit context: Context): Unit =
       errors.collectFirst {
-        case dte: DivergentImplicitTypeError => dte
+        case ite: ImplicitTypeError => ite
       } match {
-        case Some(divergent) =>
+        case Some(divergent: DivergentImplicitTypeError) =>
           // DivergentImplicit error has higher priority than "no implicit found"
           // no need to issue the problem again if we are still in silent mode
           if (context.reportErrors) {
             context.issue(divergent.withPt(paramTp))
             errorBuffer.retain {
-              case dte: DivergentImplicitTypeError => false
+              case ite: ImplicitTypeError => false
+              case _ => true
+            }
+          }
+        case Some(ite) =>
+          if (context.reportErrors) {
+            context.issue(ite)
+            errorBuffer.retain {
+              case ite: ImplicitTypeError => false
               case _ => true
             }
           }
```

Comment:
Hmm I just noticed that the cases are exactly the same except for the small `divergent.withPt(paramTp)` vs `ite` diff. I'd write something like:

```
case Some(ite) =>
  if (context.reportErrors) {
    val err = if (ite.isInstanceOf[DivergentImplicitTypeError]) divergent.withPt(paramTp) else ite
    context.issue(err)
    errorBuffer.retain {
      case ite: ImplicitTypeError => false
      case _ => true
    }
  }
```